src/pyseekdb/client/client_base.py

```
@@ -928,11 +973,16 @@ def _create_collection_meta_v2(
             collection_name_in_table = escape_string(collection_name)
-            delete_sql = f"DELETE FROM `{CollectionNames.sdk_collections_table_name()}` WHERE COLLECTION_NAME = '{collection_name_in_table}'"
-            self._execute(delete_sql)
-            insert_sql = f"INSERT INTO `{CollectionNames.sdk_collections_table_name()}` (COLLECTION_NAME, SETTINGS) VALUES ('{collection_name_in_table}', '{settings_str}')"
-            self._execute(insert_sql)
-            collection_id = self._get_collection_id(collection_name)
+            try:
+                collection_id = self._get_collection_id(collection_name)
+            except ValueError:
+                insert_sql = (
+                    f"INSERT INTO `{CollectionNames.sdk_collections_table_name()}` "
+                    f"(COLLECTION_NAME, SETTINGS) VALUES ('{collection_name_in_table}', '{settings_str}')"
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)